src/game/server/baseentity.cpp

```diff
@@ -1697,11 +1697,11 @@ int CBaseEntity::TakeDamage( const CTakeDamageInfo &inputInfo )
 		if ( info.GetAttacker() )
 		{
 			info.ScaleDamage( info.GetAttacker()->GetAttackDamageScale( this ) );
+            
+			// Scale the damage by my own modifiers
+			info.ScaleDamage( GetReceivedDamageScale( info.GetAttacker() ) );
```

Comment:
Moving this doesn't seem right; if the attacker is worldspawn/-1 the recipient should still take increased damage via their increased damage scale modifier.

Reply:
It's crashing in hl2 based mods where the damage multiplier isn't used and is basically NULL. in TF2 this change won't have any effect.

Reply:
GetReceivedDamageScale doesn't even use its *pAttacker parameter, though. Was it info.getattacker() returning null, or was info itself null? If the latter, `info ? info.GetAttacker() : NULL` is probably the way to do this, instead of moving the call.

Reply:
FYI something still doesn't seem right here, I don't see why info.GetAttacker() returning null would cause a crash but `info.GetAttacker() ? info.GetAttacker() : NULL` wouldn't, given that it'd just be `NULL ? NULL : NULL` in that case. Something else is going wrong, or that isn't actually what's crashing it.